### ISSUE TITLE
Ledger wallet not automatically getting funds

### ISSUE THREAD

**Issue by nodecheck** [btcpayserver/btcpayserver#801]
Ledger was connected to BTCPay correctly, the correct derivation was chosen, and the address was verified as correct.  An invoice was paid, and the funds show in BTCPay under wallet.  However, this balance is not shown on the ledger wallet.  Over 6 confirmations are shown for the transaction.

* Version of BTCPay Server: 1.0.3.94
* Deployment method: docker

We were expecting that once the invoice has been paid, the balance should have been shown correctly on the ledger wallet.  If it doesn't do this automatically, why is that?  Surely we shouldn't have to manually go to wallet within BTCPay and then send the amount to our hardware wallet?  I see a send option, so can do this, but seems a bit strange when all of this should have been happening automatically - either that or this is misunderstood and not how it works?

**Comment by Kukks** [comment]
https://docs.btcpayserver.org/faq-and-common-issues/faq-wallet#missing-payments-in-my-software-or-hardware-wallet

**Comment by Kukks** [comment]
TL;DR; Your funds are safu and 99% of all wallets don't care for real-world merchant use-cases.
https://github.com/LedgerHQ/ledger-live-desktop/issues/1279

**Comment by pavlenex** [comment]
The balance is not shown, because Ledger Live wallet does not support gap limit higher than 20. Your funds are there, but most of the wallets suck. This has been documented already and you even have a warning in BTCPay Page ^ shared by Kukks and also you have a warning on BTCPay wallet page.

If you have a Nano S, use it in your BTCPay, it's way more private than using it with Live since it relies on your full node, it has customizable gap limit.

**Comment by nodecheck** [comment]
We have Nano S connected to BTCPay, and it doesn't show in the wallet.

**Comment by pavlenex** [comment]
Which wallet? If it's not shown in BTCPay internal wallet, [rescan](https://docs.btcpayserver.org/features/wallet#re-scan) your wallet if you're the admin of the server.

Don't use Ledger Live if you're merchant, it's not designed for merchants.

**Comment by nodecheck** [comment]
The links you sent don't resolve the problem.  So effectively what you are saying is don't use Nano S with BTCPay because it obviously doesn't work.  I have funds now shown in BTCPay but have no way to obtain them, and the links you have provided don't explain how to gain access to it.  Your documentation and links supposedly explaining what to do are incomplete.

**Comment by Coinomatron** [comment]
Crude option would be to send a minimum payment to the 20th address so it creates 20 more addresses. Do that enough times and the used address will show up.

Electrum personal server allows you to set a gap limit (of 1000, for example), so the connected electrum client will always show received funds.

**Comment by lobsterpub** [comment]
Why can't you just use the BTCPay internal wallet?

As far as I can tell this is a Ledger Live wallet issue.  They don't look more than 20 addresses ahead if none have received funds.  As stated above, Electrum is another option.

**Comment by nodecheck** [comment]
Will try to switch back to internal wallet instead of using nano/ledger.

**Comment by Kukks** [comment]
@nodecheck You are mixing things: Ledger Live app and Ledger Nano S are two separate things. Nano S, the hardware device can be used with Ledger Live.

The issue you are experiencing is with Ledger Live, not BTCPay. You can use your Nano S with both the built-in wallet tool in BtcPay and Electrum(once you import your wallet). Your funds are not stuck and we are not telling you to not use your Nano.

I urge you to vent your frustrations on https://github.com/LedgerHQ/ledger-live-desktop/issues/1279 so that they will feel pressured and change their minds.

**Comment by nodecheck** [comment]
@Kukks OK thx.  In the meantime I decided to check/verify the xpub/derivation and addresses shown in case the nano s wasn't connected properly.  After a rescan the balance of what we had received re-appear when we went to wallet in BTCPay.  We then used the send option to send out to another address, and clicked verify on the nano device.  It worked.  We had tried this previously and it didn't work, so possibly something was not working quite as it should.

I'll join in that conversation and LedgerHQ.

**Comment by pavlenex** [comment]
There's absolutely no reason why one should use Ledger Live. 

Internal BTCPay wallet is not only better suited for your use case and eliminates the gap limit problem, but also relies on your full node for maximum privacy/security.

Glad you resolved it via the workaround, but still, there's no reason to do it if you use internal wallet or Electrum.
